diff --git a/test/common.rb b/test/common.rb
index 3b179a7..53430e7 100644
--- a/test/common.rb
+++ b/test/common.rb
@@ -23,8 +23,6 @@ module LZWS
       ]
       .each { |path| FileUtils.touch path }
 
-      PORT = 54_010
-
       ENCODINGS = %w[
         binary
         UTF-8
diff --git a/test/stream/reader.test.rb b/test/stream/reader.test.rb
index 5708779..a061908 100644
--- a/test/stream/reader.test.rb
+++ b/test/stream/reader.test.rb
@@ -20,7 +20,6 @@ module LZWS
         String = LZWS::String
 
         ARCHIVE_PATH      = Common::ARCHIVE_PATH
-        PORT              = Common::PORT
         ENCODINGS         = Common::ENCODINGS
         TRANSCODE_OPTIONS = Common::TRANSCODE_OPTIONS
         TEXTS             = Common::TEXTS
@@ -264,7 +263,7 @@ module LZWS
         end
 
         def test_readpartial
-          ::TCPServer.open PORT do |server|
+          ::TCPServer.open 0 do |server|
             TEXTS.each do |text|
               PORTION_LENGTHS.each do |portion_length|
                 [true, false].map do |with_buffer|
@@ -285,7 +284,7 @@ module LZWS
                       prev_result       = "".b
                       decompressed_text = "".b
 
-                      ::TCPSocket.open "localhost", PORT do |socket|
+                      ::TCPSocket.open "localhost", server.addr[1] do |socket|
                         instance = target.new socket, decompressor_options
 
                         begin
@@ -321,7 +320,7 @@ module LZWS
         # -- asynchronous --
 
         def test_read_nonblock
-          ::TCPServer.open PORT do |server|
+          ::TCPServer.open 0 do |server|
             TEXTS.each do |text|
               PORTION_LENGTHS.each do |portion_length|
                 get_compressor_options do |compressor_options|
@@ -350,7 +349,7 @@ module LZWS
 
                     decompressed_text = "".b
 
-                    ::TCPSocket.open "localhost", PORT do |socket|
+                    ::TCPSocket.open "localhost", server.addr[1] do |socket|
                       instance = target.new socket, decompressor_options
 
                       begin
diff --git a/test/stream/writer.test.rb b/test/stream/writer.test.rb
index eea1551..ff9a3d3 100644
--- a/test/stream/writer.test.rb
+++ b/test/stream/writer.test.rb
@@ -19,7 +19,6 @@ module LZWS
         String = LZWS::String
 
         ARCHIVE_PATH      = Common::ARCHIVE_PATH
-        PORT              = Common::PORT
         ENCODINGS         = Common::ENCODINGS
         TRANSCODE_OPTIONS = Common::TRANSCODE_OPTIONS
         TEXTS             = Common::TEXTS
@@ -155,7 +154,7 @@ module LZWS
         # -- asynchronous --
 
         def test_write_nonblock
-          ::TCPServer.open PORT do |server|
+          ::TCPServer.open 0 do |server|
             TEXTS.each do |text|
               PORTION_LENGTHS.each do |portion_length|
                 sources = get_sources text, portion_length
@@ -180,7 +179,7 @@ module LZWS
                       end
                     end
 
-                    TCPSocket.open "localhost", PORT do |socket|
+                    TCPSocket.open "localhost", server.addr[1] do |socket|
                       instance = target.new socket, compressor_options
 
                       begin
