diff --git a/ext/extconf.rb b/ext/extconf.rb
index 0f0f282..a2ec6b6 100644
--- a/ext/extconf.rb
+++ b/ext/extconf.rb
@@ -75,6 +75,13 @@ $srcs = %w[
 .map { |name| "src/#{extension_name}/#{name}.c" }
 .freeze
 
+# Removing library duplicates.
+$libs = $libs.split(%r{\s})
+  .reject(&:empty?)
+  .sort
+  .uniq
+  .join " "
+
 if ENV["CI"] || ENV["COVERAGE"]
   $CFLAGS << " --coverage"
   $LDFLAGS << " --coverage"
