diff --git a/ext/extconf.rb b/ext/extconf.rb
index 583131f..42ea4b5 100644
--- a/ext/extconf.rb
+++ b/ext/extconf.rb
@@ -61,6 +61,13 @@ $srcs = %w[
 .map { |name| "src/#{extension_name}/#{name}.c" }
 .freeze
 
+# Removing library duplicates.
+$libs = $libs.split(%r{\s})
+  .reject(&:empty?)
+  .sort
+  .uniq
+  .join " "
+
 $CFLAGS << " -Wno-declaration-after-statement"
 $VPATH << "$(srcdir)/#{extension_name}:$(srcdir)/#{extension_name}/stream"
 # rubocop:enable Style/GlobalVars
