diff --git a/test/common.rb b/test/common.rb
index 3df084c..06154d0 100644
--- a/test/common.rb
+++ b/test/common.rb
@@ -23,9 +23,6 @@ module LZWS
       ]
       .each { |path| FileUtils.touch path }

-      # Port will be changed each 20 seconds.
-      PORT = 51_000 + (Time.now.to_i / 20) % 1000
-
       ENCODINGS = %w[
         binary
         UTF-8
diff --git a/test/stream/reader.test.rb b/test/stream/reader.test.rb
index e033f28..625e908 100644
--- a/test/stream/reader.test.rb
+++ b/test/stream/reader.test.rb
@@ -20,7 +20,6 @@ module LZWS
         String = LZWS::String

         ARCHIVE_PATH          = Common::ARCHIVE_PATH
-        PORT                  = Common::PORT
         ENCODINGS             = Common::ENCODINGS
         TRANSCODE_OPTIONS     = Common::TRANSCODE_OPTIONS
         TEXTS                 = Common::TEXTS
@@ -498,7 +497,7 @@ module LZWS
         # -- server --

         protected def start_server(&block)
-          ::TCPServer.open PORT, &block
+          ::TCPServer.open 0, &block
         end

         protected def server_nonblock_test(server, text, compressor_options = {}, decompressor_options = {}, &_block)
@@ -525,7 +524,7 @@ module LZWS
           end

           decompressed_text =
-            ::TCPSocket.open "localhost", PORT do |socket|
+            ::TCPSocket.open "localhost", server.addr[1] do |socket|
               instance = target.new socket, decompressor_options

               begin
diff --git a/test/stream/writer.test.rb b/test/stream/writer.test.rb
index 13f4985..669b2bd 100644
--- a/test/stream/writer.test.rb
+++ b/test/stream/writer.test.rb
@@ -20,7 +20,6 @@ module LZWS
         String = LZWS::String

         ARCHIVE_PATH          = Common::ARCHIVE_PATH
-        PORT                  = Common::PORT
         ENCODINGS             = Common::ENCODINGS
         TRANSCODE_OPTIONS     = Common::TRANSCODE_OPTIONS
         TEXTS                 = Common::TEXTS
@@ -391,7 +390,7 @@ module LZWS
         # -- server --

         protected def start_server(&block)
-          ::TCPServer.open PORT, &block
+          ::TCPServer.open 0, &block
         end

         protected def server_nonblock_test(server, text, portion_length, compressor_options = {}, decompressor_options = {}, &_block)
@@ -413,7 +412,7 @@ module LZWS
             end
           end

-          TCPSocket.open "localhost", PORT do |socket|
+          TCPSocket.open "localhost", server.addr[1] do |socket|
             instance = target.new socket, compressor_options

             begin
