diff --git a/test/common.rb b/test/common.rb
index 730c543..fdb377d 100644
--- a/test/common.rb
+++ b/test/common.rb
@@ -18,8 +18,6 @@ module ZSTDS
       ]
       .each { |path| FileUtils.touch path }

-      PORT = 54_030
-
       ENCODINGS = %w[
         binary
         UTF-8
diff --git a/test/stream/reader.test.rb b/test/stream/reader.test.rb
index 179f953..fefcfe7 100644
--- a/test/stream/reader.test.rb
+++ b/test/stream/reader.test.rb
@@ -20,7 +20,6 @@ module ZSTDS
         String = ZSTDS::String

         ARCHIVE_PATH      = Common::ARCHIVE_PATH
-        PORT              = Common::PORT
         ENCODINGS         = Common::ENCODINGS
         TRANSCODE_OPTIONS = Common::TRANSCODE_OPTIONS
         TEXTS             = Common::TEXTS
@@ -264,7 +263,7 @@ module ZSTDS
         end

         def test_readpartial
-          ::TCPServer.open PORT do |server|
+          ::TCPServer.open 0 do |server|
             TEXTS.each do |text|
               PORTION_LENGTHS.each do |portion_length|
                 [true, false].map do |with_buffer|
@@ -285,7 +284,7 @@ module ZSTDS
                       prev_result       = "".b
                       decompressed_text = "".b

-                      ::TCPSocket.open "localhost", PORT do |socket|
+                      ::TCPSocket.open "localhost", server.addr[1] do |socket|
                         instance = target.new socket, decompressor_options

                         begin
@@ -321,7 +320,7 @@ module ZSTDS
         # -- asynchronous --

         def test_read_nonblock
-          ::TCPServer.open PORT do |server|
+          ::TCPServer.open 0 do |server|
             TEXTS.each do |text|
               PORTION_LENGTHS.each do |portion_length|
                 get_compressor_options do |compressor_options|
@@ -350,7 +349,7 @@ module ZSTDS

                     decompressed_text = "".b

-                    ::TCPSocket.open "localhost", PORT do |socket|
+                    ::TCPSocket.open "localhost", server.addr[1] do |socket|
                       instance = target.new socket, decompressor_options

                       begin
diff --git a/test/stream/writer.test.rb b/test/stream/writer.test.rb
index fcff69c..0b7d28f 100644
--- a/test/stream/writer.test.rb
+++ b/test/stream/writer.test.rb
@@ -19,7 +19,6 @@ module ZSTDS
         String = ZSTDS::String

         ARCHIVE_PATH      = Common::ARCHIVE_PATH
-        PORT              = Common::PORT
         ENCODINGS         = Common::ENCODINGS
         TRANSCODE_OPTIONS = Common::TRANSCODE_OPTIONS
         TEXTS             = Common::TEXTS
@@ -161,7 +160,7 @@ module ZSTDS
         # -- asynchronous --

         def test_write_nonblock
-          ::TCPServer.open PORT do |server|
+          ::TCPServer.open 0 do |server|
             TEXTS.each do |text|
               PORTION_LENGTHS.each do |portion_length|
                 sources = get_sources text, portion_length
@@ -186,7 +185,7 @@ module ZSTDS
                       end
                     end

-                    TCPSocket.open "localhost", PORT do |socket|
+                    TCPSocket.open "localhost", server.addr[1] do |socket|
                       instance = target.new socket, compressor_options

                       begin
